autologin/app.py

```diff
@@ -10,20 +10,51 @@
 static_dir = os.path.join(server_path, 'static')
 browser_dir = os.path.join(static_dir, 'browser')
 
-# Read config (defaults are in autologin.conf)
-config = configparser.ConfigParser()
-# Read default config
-config.read(os.path.join(server_path, 'autologin.cfg'))
-# Override by user-supplied config
-config.read([
-    os.path.expanduser('~/.autologin.cfg'),
-    '/etc/autologin.cfg',
-    ])
-
-# Initiate flask app
-app = Flask(__name__)
-app.secret_key = config.get('autologin', 'secret_key')
-app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///{}'.format(os.path.abspath(
-    os.path.join(config.get('autologin', 'db_folder'), 'db.sqlite')))
-app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+
+def init_app():
+    # Read config (defaults are in autologin.conf)
+    config = configparser.ConfigParser()
+    # Read default config
+    default_config = os.path.join(server_path, 'autologin.cfg')
+    config.read(default_config)
+    # Override by user-supplied config
+    user_configs = [
+        os.path.expanduser('~/.autologin.cfg'),
+        '/etc/autologin.cfg',
+        ]
+    config.read(user_configs)
+    used_config = default_config
+    for path in user_configs:
+        if os.path.exists(path):
+            used_config = path
+            break
+
+    # Initiate flask app
+    app = Flask(__name__)
+    app.secret_key = config.get('autologin', 'secret_key')
+    try:
+        db_path = config.get('autologin', 'db')
+    except configparser.NoOptionError:
+        db_path = os.path.join(os.path.dirname(__file__), 'db.sqlite')
+    else:
+        if not os.path.isabs(db_path):
+            raise RuntimeError(
+                'You must specify an absolute path to the database. '
+                'Invalid relative path "{db_path}" in config file {used_config}'
+                .format(db_path=db_path, used_config=used_config))
```

Comment:
Fixed, also noticed that ``config.read`` returns a list of filenames read, which is what we need here.
Thanks! :)